# Python Basic Assignment solution

### Workshop 19.03.2019

Go through the exercises and solve the tasks.

#### Example exercise

What is 43+67?

In [1]:
43+67







110

Solution: 110

___

<h3 align="center">Exercise 1</h3> 

What is 8 to the power of 3? 

In [2]:
# Write code here..


8**3




512

Solution: 512

___

<h3 align="center">Exercise 2</h3> 

Create a function that prints the input

In [3]:
# Write code here..

def my_print_function(my_input):
    
    print(my_input)
    
    
    

Solution: Run this code and se if the function works

In [4]:
my_print_function("Yes! \nThis works!")

Yes! 
This works!


___

<h3 align="center">Exercise 3</h3> 

Split the string:  

**s = " this is a string"  **

into a list

In [5]:
# Write code here..

s= "this is a string".split()

s

['this', 'is', 'a', 'string']

Solution:

[ 'this', 'is', 'a', 'string']

___

<h3 align="center">Exercise 4</h3> 

#### Excerise 4

Write a function that inputs height and weight and returns BMI.

**Formel: **
\begin{align}
BMI & = \frac{weight[kg]}{height[m]^2} 
\end{align}

In [6]:
# Write code here..

def bmi(height,weight):
    
    return weight/height**2




Solution: test with 1,85 m and 75 kg --> Should be a BMI of  aprox **21.9**

In [7]:
'''Test your code here...'''

bmi(1.85,75)


21.913805697589478

___

<h3 align="center">Exercise 5</h3> 

#### Excerise 5 - Learn about list indexing

Pick out your birthday number from the list of days and months and print it.

In [8]:
# Run this code first

'''Part 1'''
date_list = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31]
month_list = ['jan', 'feb', 'mar', 'apr', 'mai', 'jun', 'jul', 'aug', 'sep', 'okt', 'nov', 'des']


'''Part 2'''
date_list_hard = [[1,2,3,4,5,6,7,8,9,],[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26],[27,28,29,30,31]]
month_list_hard = [[['jan', 'feb'] ,'mar', 'apr'],['mai', 'jun'],['jul', 'aug', 'sep', 'okt', 'nov', 'des']]



In [9]:
# Write your code here..

''' My birthday is 16.feb'''

date = date_list[15]
month = month_list[1]
print("My birthday is {}.{}".format(date, month))


date_hard = date_list_hard[1][6]
month_hard = month_list_hard[0][0][1]
print("My birthday is {}.{}".format(date_hard, month_hard))

My birthday is 16.feb
My birthday is 16.feb


Solution example: "My birthday is 16.feb"

___

<h3 align="center">Exercise 6</h3> 

#### Excerise 6 - Learn about the map function

Convert a list of heights and weight to a list of BMI-values. 

TIPS:

"map(function_to_apply, list_of_inputs)" applies a function to all the items in an input list. 

In [10]:
# Write code here..

'''(weight, height) touples'''
my_list_of_personal_information = [(75,1.86),(121,1.98),(53,1.46),(84,1.79)]



In [11]:
'''Alternative 1'''
# input = touple --> (vekt,  høyde)
def bmi_adapted(touple):
    
    return touple[0]/touple[1]**2


answer = list(map(bmi_adapted,my_list_of_personal_information))
print(answer)

'''Alternative 2'''
#Create empty list
list_of_bmi = []

for each_touple in my_list_of_personal_information:
    
    current_bmi = bmi(each_touple[1], each_touple[0])
    
    list_of_bmi.append(current_bmi)
    
print(list_of_bmi)

[21.678806798473808, 30.8641975308642, 24.86395196096829, 26.216410224399986]
[21.678806798473808, 30.8641975308642, 24.86395196096829, 26.216410224399986]


Solution:

[21.678806798473808, 30.8641975308642, 24.86395196096829, 26.216410224399986]

___

<h3 align="center">Exercise 7</h3> 

#### Excerise 7

Make two seperatee list from the email list. 

One with all the names and one with all the companies.

In [12]:
# Write code here..

email_list = ['ola@facebook.com','tine@google.com','josef@amazon.com', 'tom@xxl.com']

def get_names(input_string):
    
    name = input_string.split('@')[0]
    
    return name

def get_company(input_string):
    
    company_mail = input_string.split('@')[1]
    
    company = company_mail.split('.')[0]
    
    return company

names = list (   map(get_names,email_list)    )
companies = list(   map(get_company,email_list)  )


Solution

In [13]:
''' Test your code here'''

print(names)
print(companies)

['ola', 'tine', 'josef', 'tom']
['facebook', 'google', 'amazon', 'xxl']


# Well done!